apps/meteor/app/livechat/server/lib/guests.ts

```diff
@@ -78,11 +78,32 @@ export async function saveGuest(
 	return ret;
 }
 
-export async function removeGuest({ _id, token }: { _id: string; token: string }) {
-	await cleanGuestHistory(token);
+async function removeGuest({ _id }: { _id: string }) {
+	await cleanGuestHistory(_id);
 	return LivechatVisitors.disableById(_id);
 }
 
+export async function removeContactsByVisitorId({ _id }: { _id: string }) {
+	// A visitor shouldn't have many contacts associated, so we can remove them like this
+	const contacts = await LivechatContacts.findAllByVisitorId(_id).toArray();
+	if (!contacts.length) {
+		livechatLogger.debug({ msg: 'No contacts found for visitor', visitorId: _id });
+		await removeGuest({ _id });
+	}
+
+	// And a contact shouldn't have many channels associated, so we can do this
+	livechatLogger.debug({ msg: 'Removing channels for contacts', visitorId: _id, contacts: contacts.map(({ _id }) => _id) });
+	for await (const contact of contacts) {
+		for await (const { visitor } of contact.channels) {
+			await removeGuest({ _id: visitor.visitorId });
+		}
+
+		await LivechatContacts.disableByVisitorId(_id);
+	}
+
+	return true;
```

Comment:
Any reason for returning a boolean?

Reply:
It's actually to keep this error on the endpoint:
```
		const { _id } = visitor;
		const result = await removeContactsByVisitorId({ _id });
		if (!result) {
			throw new Meteor.Error('error-removing-visitor', 'An error ocurred while deleting visitor');
		}